crates/tx5/tests/tests/flaky_sig.rs

```diff
@@ -0,0 +1,285 @@
+                                                tracing::debug!("Downstream connection closed");
+                                                break;
+                                            }
+                                            tracing::debug!("Received {} bytes from downstream", amt);
+                                            if let Err(e) = socket.write(&tx_buf[..amt]).await {
+                                                tracing::error!("Failed to write to peer: {}", e);
+                                                break;
+                                            }
+                                        }
+                                        Err(e) => {
+                                            tracing::error!("Failed to read from downstream: {}", e);
+                                            break;
+                                        }
+                                    }
+                                }
+                            }
+                        }
+
+                        tracing::info!("Connection closed");
+                    }).abort_handle();
+
+                    inner.lock().await.push(relay_task);
+                    tracing::debug!("Relay task spawned for connection");
+                }
+            }
+        }).abort_handle();
+
+        let bound_addr = addr_rx.await.unwrap();
+
+        Self {
+            listen_task,
+            inner,
+            bound_addr,
+        }
+    }
+
+    async fn just_lose_it(&self) {
+        tracing::debug!(
+            "Losing connections and fulfilling my purpose as a flaky relay"
+        );
+        let mut inner = self.inner.lock().await;
+        for task in inner.drain(..) {
+            tracing::debug!("Aborting a connection task");
+            task.abort();
+        }
+    }
+
+    async fn just_lose_one(&self) {
+        tracing::debug!(
+            "Losing one connection and fulfilling my purpose as a flaky relay"
+        );
+        let mut inner = self.inner.lock().await;
+        if let Some(task) = inner.pop() {
+            task.abort();
+            tracing::debug!("One connection lost");
+        } else {
+            tracing::warn!("No connection to lose");
+        }
+    }
+}
+
+async fn ep_for_relay(
+    bound_addr: std::net::SocketAddr,
+) -> (PeerUrl, Endpoint, EndpointRecv) {
+    let config = tx5::Config {
+        signal_allow_plain_text: true,
+        timeout: std::time::Duration::from_secs(5),
+        ..Default::default()
+    };
+
+    let (ep, recv) = Endpoint::new(Arc::new(config));
+    let sig = format!("ws://{}", bound_addr);
+    let peer_url = ep.listen(tx5::SigUrl::parse(sig).unwrap()).await.unwrap();
+    (peer_url, ep, recv)
+}
+
+async fn wait_for_connected(
+    e1: &Endpoint,
+    e2: &Endpoint,
+    r1: &mut EndpointRecv,
+    r2: &mut EndpointRecv,
+) {
+    // Now we expect both peers to be able to reconnect and send messages again. They might have
```

Comment:
I'm wondering whether we should move this comment outside of the function to where the function is being called?

Reply:
Good point, this function was extracted as common code and the comment should have become a doc comment